packages/ec-core/src/distributions/wrappers/owned.rs

```diff
@@ -2,12 +2,13 @@ use std::{borrow::Borrow, marker::PhantomData, num::NonZeroUsize};
 
 use rand::{distributions::Uniform, prelude::Distribution};
 
-use crate::distributions::wrappers::slice_cloning::EmptySlice;
+use crate::distributions::{choices::ChoicesDistribution, wrappers::slice_cloning::EmptySlice};
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Eq)]
 pub struct OneOfCloning<T, U> {
     collection: T,
     range: Uniform<usize>,
+    num_choices: NonZeroUsize,
```

Comment:
The relationship between these three fields needs to be clearly documented, and we should make sure we have tests (possibly doctests) that confirm that after calling `::new()` the relationships between these fields are preserved.